compiler/src/dotty/tools/dotc/config/CompilerCommand.scala

```diff
@@ -71,7 +71,14 @@ object CompilerCommand extends DotClass {
       val ss                  = (ctx.settings.allSettings filter cond).toList sortBy (_.name)
       val width               = (ss map (_.name.length)).max
       def format(s: String)   = ("%-" + width + "s") format s
-      def helpStr(s: Setting[_]) = s"${format(s.name)} ${s.description}"
+      def formatSettings(name: String, value: String) = if (value.nonEmpty) format("\n") ++ s"  $name: $value." else ""
+      def helpStr(s: Setting[_]) = {
+        val help = StringBuilder.newBuilder
+        help.append(s"${format(s.name)} ${s.description}")
+        help.append(formatSettings("Default", s.defaultValue))
+        help.append(formatSettings("Choices", s.legalChoices))
+        help.toString
```

Comment:
Since this code already using the `s` formatter, what about just one expression?
`s"${format(s.name)} ${s.description}${formatSettings("Default", s.defaultValue)}${formatSettings("Choices", s.legalChoices)}"`? You can put arbitrary expressions inside `${}` 🙂